x/evm/genesis.go

```diff
@@ -77,19 +77,27 @@ func InitGenesis(
 func ExportGenesis(ctx sdk.Context, k *keeper.Keeper, ak types.AccountKeeper) *types.GenesisState {
 	var ethGenAccounts []types.GenesisAccount
 	ak.IterateAccounts(ctx, func(account authtypes.AccountI) bool {
```

Comment:
actually this should iterate only through the contracts and not the accounts. We should have a reverse key from `code_hash -> []address` to be able to iterate through all the contracts without having to touch the accounts that are EOAs